lib/viewTransitions.d.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+interface Document {
+  startViewTransition(fn: () => void): ViewTransition;
+}
+  
+interface ViewTransition {
+  ready: Promise<void>;
+}
```

Comment:
These declarations allow us to avoid `@ts-expect-error` or `@ts-ignore` in documentation examples. `document.startViewTransition` is included in TS 5.6 so this should go away when we update. Not sure about the React one.